app/views/onboarding/Onboarding4.js

```diff
@@ -17,13 +19,61 @@ import languages from '../../locales/languages';
 const width = Dimensions.get('window').width;
 
 const Onboarding = props => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [signedEula, setSignedEula] = useState(false);
   return (
     <View style={styles.mainContainer}>
       <StatusBar
         barStyle='dark-content'
         backgroundColor='transparent'
         translucent={true}
       />
+      <View style={styles.centeredView}>
+        <Modal
+          animationType='slide'
+          transparent={true} // eslint-disable-line react/jsx-boolean-value
+          visible={modalVisible}
+          onRequestClose={() => {
+            setModalVisible(false);
+          }}>
+          <View
```

Comment:
It's ok for the Modal activation state to be in this file. But this is a component in its own right - with its own styles and state. 

Make it a new file.